homeassistant/components/connectedcars/__init__.py

```diff
@@ -0,0 +1,129 @@
+    ATTR_API_VEHICLE_MODEL,
+    ATTR_API_VEHICLE_NAME,
+    ATTR_API_VEHICLE_ODOMETER,
+    ATTR_API_VEHICLE_POS_LATITUDE,
+    ATTR_API_VEHICLE_POS_LONGITUDE,
+    ATTR_API_VEHICLE_VIN,
+    ATTR_API_VEHICLE_VOLTAGE,
+    CONF_NAMESPACE,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor", "device_tracker"]
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+UPDATE_INTERVAL = timedelta(minutes=5)  # Maybe create a config for this
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the ConnectedCars.io component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry):
+    """Set up ConnectedCars.io from a config entry."""
+    username = config_entry.data[CONF_USERNAME]
+    password = config_entry.data[CONF_PASSWORD]
+    namespace = config_entry.data[CONF_NAMESPACE]
+    ccah = ConnectedCarsClient(username, password, namespace)
+
+    coordinator = ConnectedCarsDataUpdateCoordinator(hass, ccah, UPDATE_INTERVAL)
+    await coordinator.async_refresh()
+
+    if not coordinator.last_update_success:
+        raise ConfigEntryNotReady
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][config_entry.entry_id] = coordinator
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(config_entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass, config_entry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(config_entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(config_entry.entry_id)
+
+    return unload_ok
+
+
+class ConnectedCarsDataUpdateCoordinator(DataUpdateCoordinator):
+    """Define an object to hold Connected Cars data."""
+
+    def __init__(self, hass, ccah: ConnectedCarsClient, update_interval):
+        """Initialize."""
+        self.ccah = ccah
+        super().__init__(hass, _LOGGER, name=DOMAIN, update_interval=update_interval)
+
+    async def _async_update_data(self):
+        """Update data via library."""
+        with async_timeout.timeout(20):
+            try:
+                response = await self.ccah.async_query(QUERY_COMPLETE)
+            except ConnectedCarsException as error:
+                raise UpdateFailed(error) from error
+
+        data = response["data"]
+        viewer = data["viewer"]
+        vehicles = viewer["vehicles"]
+        vehicle = vehicles[0]["vehicle"]
+
+        flattened_data = {
+            ATTR_API_USER_FIRSTNAME: viewer["firstname"],
+            ATTR_API_USER_LASTNAME: viewer["lastname"],
+            ATTR_API_USER_EMAIL: viewer["email"],
+            ATTR_API_VEHICLE_ID: vehicle["id"],
+            ATTR_API_VEHICLE_VIN: vehicle["vin"],
```

Comment:
The config asks for a vin, but I don't see it being used from the entry data in this part. How does this work on getting the right vehicle?

Reply:
It does not matter currently, it is something I wanted to add support for (multiple cars) after this initial add was done. But until I am ready for that, I guess I should remove this.